cmd/cloudinit/cloudinit.go

```diff
@@ -230,6 +231,7 @@ func saveCloudConfig() error {
 	}
 
 	userData := string(userDataBytes)
+	userDataBytes = substituteVars(userData, metadata)
```

Comment:
Strings are immutable, so `userData` is left unchanged. It's probably not what you want.